CMakeLists.txt

```diff
@@ -50,18 +49,7 @@ include(${CMAKE_CURRENT_LIST_DIR}/cmake/Modules/MaCh3Dependencies.cmake)
 
 ############################  C++ Compiler  ####################################
 if (NOT DEFINED CMAKE_CXX_STANDARD OR "${CMAKE_CXX_STANDARD} " STREQUAL " ")
-  set(CMAKE_CXX_STANDARD 14)
-endif()
-
-# KS: If C++ standard is lower than C++ standard used for ROOT compilation things will go terribly wrong
-if(DEFINED ROOT_CXX_STANDARD AND ROOT_CXX_STANDARD GREATER CMAKE_CXX_STANDARD)
-  set(CMAKE_CXX_STANDARD ${ROOT_CXX_STANDARD})
-endif()
-cmessage(STATUS "CMAKE CXX Standard: ${CMAKE_CXX_STANDARD}")
-
-# KS: ROOT changed cmake in 6.32, we should move avay from using Luke's hack, keep it for due to compatibility
-if(ROOT_CXX_STANDARD LESS 14 AND ROOT_VERSION VERSION_LESS 6.32.00)
-  cmessage(WARNING "ROOT CXX STANDARD: ${ROOT_CXX_STANDARD}")
+  set(CMAKE_CXX_STANDARD 17)
```

Comment:
can you try ussing ROOT_CXX_STANDARD?
It is in root cofnig
https://github.com/root-project/root/blob/8d5764957e4bcb9cdc0bca44f83fb2fb4fe1815d/cmake/scripts/RootUseFile.cmake.in#L12

Reply:
yep, done

Reply:
oh, no its not, if we 'use' the RootUseFile then it sets the standard for the whole compilation, but it doesn't expose directly what standard it uses. I'll just grab the flag-stripping code from NuHepMC

Reply:
I've put something a bit hacky in for the moment. NuOscillator uses the nuhepmc/cmakemodules still and it ends up setting up root, so I'd need to PR it out of that to properly just let ROOT set itself up. But this builds for the moment